src/vs/base/parts/tree/browser/treeModel.ts

```diff
@@ -959,7 +959,9 @@ export class TreeModel {
 		return this.input ? this.input.getElement() : null;
 	}
 
-	public refresh(element: any = null, recursive: boolean = true): Promise<any> {
+	public async refresh(element: any = null, recursive: boolean = true): Promise<any> {
+		// Add loading spinner for item during refresh process.
+		this.addTraits('loading', [element]);
```

Comment:
Shouldn't this go below the `if (!item)` check on 967? If we don't have an item there's nothing we should be adding the trait to.